### ISSUE TITLE
Unused snapshot falsely detected when tests have same names

### ISSUE THREAD

**Issue by daphtdazz** [tophat/syrupy#529]
**Describe the bug**

If a test in one test file has the same name as a test in another test file, and both try to do a snapshot, if we run pytest but deselect one of those tests then pytest fails erroneously saying that a snapshot is obsolete.

**To reproduce**

Create two test files (`test_a.py` and `test_b.py`) with the following contents:

```py
def test_name(snapshot):
    assert snapshot == [1, 2, 3]


def test_other():
    assert True
```

Create the snapshots:

```
pytest --snapshot-update test_a.py test_b.py
```

Attempt to run all but `test_b.py::test_name`:

```
pytest test_a.py test_b.py::test_other
```

*Result:* pytest exits with status 1 saying:

```
--------------------------- snapshot report summary ---------------------------
1 snapshot passed. 1 snapshot unused.

Re-run pytest with --snapshot-update to delete unused snapshots.
```

**Expected behavior**

pytest should exit successfully and we shouldn't be told a snapshot is unused, because we didn't run the test that used it.

**Screenshots**

Should be clear from above.

**Environment (please complete the following information):**

- OS: several (macOS, Linux)
- Syrupy Version: 1.3.1
- Python Version: 3.6 and 3.7 (and I assume all the others)

**Additional context**

None

**Comment by noahnu** [comment]
Thanks for reporting this issue.

I have a fix in #531, just validating the correctness.

**Comment by noahnu** [comment]
Unfortunately my first attempt didn't work. This is difficult since we make no assumptions around the extension being used (ambr vs. another extension). I've updated my PR with some more test coverage which has at least given me confidence that my first approach doesn't work. I'll give this some more thought.
